app/services/katello/pulp3/migration_plan.rb

```diff
@@ -105,7 +105,15 @@ def target_link_map
       end
 
       def name_for_content_view(content_view, root_repo)
-        "#{content_view.label}-#{root_repo.label}"
+        name = "#{content_view.label}-#{root_repo.label}"
+
+        if Katello::RootRepository.where(:label => root_repo.label).group(:label).count(:label)[root_repo.label] > 1
+          repo_query = Katello::Repository.joins(:root, :content_view_version => :content_view).
+              where("#{::Katello::RootRepository.table_name}.label" => root_repo.label, "#{::Katello::ContentView.table_name}.label" => content_view.label)
+          duplicate_count = repo_query.group(:library_instance_id, :content_view_id).count.select { |k,v| v > 1 }.count
```

Comment:
You are grouping by content_view_id here. So you expect multiple content views to have the same label, too? If that was possible (it is in the database) this would further complicate things.

Reply:
Its possible across organizations, this was partially why i went with root_id, as that should be completely unique.  I do need to test more this scenario!

Reply:
I'm grouping by content view and library instance_id to 'collapse' all the same ones together with that same pairing.